### PR TITLE
Refactor cloud init parser to use less dependencies. only bash and jq

### PR DESCRIPTION
**Pull request by MoeMahhouk**
In this PR, I've tackled the following:
1- Reduced the TCB but not relying on python as dependency and only parse the config with bash and jq
2- Limit attack surface by increasing the parsing security to only allow exporting a allow-listed variables only and added some format synthesizing checks.
3- Added the config-parser dependency to the reth recipe since reth binary also needs it
4- Added a cloud init URL variable such that it can be used to pass the real azure cloud init endpoint during building. either through, local.conf, directly while building our through sourcing cloud-init-conf file.

### CODE REVIEW

File: recipes-nodes/config-parser/fetch-config.sh
Comment (fnerdman) [inline recipes-nodes/config-parser/fetch-config.sh:1]: This is the url for azures config endpoint:
http://169.254.169.254/metadata/instance/compute/userData?api-version=2021-01-01&format=text
so to match this, the static `/config.json` part of the url should be removed

File: recipes-nodes/config-parser/config_parser.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    case "$1" in
+        0x[0-9a-fA-F]*) return 0 ;;
+        *) return 1 ;;
+    esac
+}
+
+# Function to validate URL
+validate_url() {
+    case "$1" in
+        http://*|https://*) return 0 ;;
```
Comment (fnerdman) [inline recipes-nodes/config-parser/config_parser.sh:37]: I don't trust the wildcard expression here, let's mark it as review required

File: recipes-nodes/config-parser/fetch-config.sh
Comment (MoeMahhouk) [reply recipes-nodes/config-parser/fetch-config.sh:1]: ops, yeah. that was supposed to be completely part of the config URL and nothing is suffixed.
I will fix that shortly.

Good catch!

File: recipes-nodes/config-parser/config_parser.sh (hunk 1 above)
Comment (MoeMahhouk) [reply recipes-nodes/config-parser/config_parser.sh:37]: I agree, we can add more restrictions or checks for it.
Currently, that function is only used once for the BuilderBeaconEndpoints variable to make sure that it starts with a valid URL and not setting arbitrary data in there.
But sure, we can synthesis the logic even more

**Pull request merged**